Our.Umbraco.TagHelpers/ImgTagHelper.cs

```diff
@@ -314,12 +314,63 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
             // The configuration allows us to define whether images are configured "mobile first". This simply alternates between min-width & max-width media queries.
             var imageSizes = GetImageSizes(MediaItem != null);
 
+            // Avoid rendering a WebP alternative if the image is a GIF
+            var imageFormat = MediaItem != null ? MediaItem.Url().Split('.').Last() : FileSource?.Split('.').Last(); 
+            var renderWebP = imageFormat != "gif";
+
             if (imageSizes != null && imageSizes.Any())
             {
                 var sb = new StringBuilder();
                 sb.AppendLine("<picture>");
 
                 imageSizes = _globalSettings.OurImg.MobileFirst ? imageSizes.OrderByDescending(o => o.ScreenSize).ToList() : imageSizes.OrderBy(o => o.ScreenSize).ToList();
+
+                if (renderWebP)
+                {
+                    foreach (var size in imageSizes)
+                    {
+                        var minWidth = size.ScreenSize switch
+                        {
+                            OurScreenSize.ExtraExtraLarge => _globalSettings.OurImg.MediaQueries.ExtraExtraLarge,
+                            OurScreenSize.ExtraLarge => _globalSettings.OurImg.MediaQueries.ExtraLarge,
+                            OurScreenSize.Large => _globalSettings.OurImg.MediaQueries.Large,
+                            OurScreenSize.Medium => _globalSettings.OurImg.MediaQueries.Medium,
+                            OurScreenSize.Small => _globalSettings.OurImg.MediaQueries.Small,
+                            _ => 0
+                        };
+
+                        double sourceHeight = 0;
+
+                        if (MediaItem != null)
+                        {
+                            #region Configure crops which can be set at variant level or inherit from the crop alias defined on the main img element itself. If neither have a crop alias, then don't use crops.
```

Comment:
I would remove this and make it a comment as the file doesn't use `#region` elsewhere.

Reply:
Good point, although this was in the original repo so will leave it as is. Not sure why it's being marked as changed.